lib/oban.ex

```diff
@@ -471,22 +471,9 @@ defmodule Oban do
     |> Notifier.notify(:signal, %{action: :stop, queue: queue})
   end
 
-  @doc """
-  Kill an actively executing job and mark it as `discarded`, ensuring that it won't be retried.
-
-  If the job happens to fail before it can be killed the state is set to `discarded`. However,
-  if it manages to complete successfully then the state will still be `completed`.
-
-  ## Example
-
-  Kill a long running job with an id of `1`:
-
-      Oban.kill_job(1)
-      :ok
-  """
-  @doc since: "0.2.0"
-  @doc deprecated: "Use Oban.cancel_job/1 instead"
-  @spec kill_job(name :: atom(), job_id :: pos_integer()) :: :ok
+  @doc false
+  @deprecated "Use Oban.cancel_job/1 instead"
```

Comment:
I just noticed [here](https://github.com/sorentwo/oban/commit/c0b84efd4bc32c36038f8653f9687dde888f5001) that you _soft-deprecated_ this function on Apr 9, so, let me know if it's too soon for a _hard-deprecation_.

Reply:
There hasn't been a release since this was soft-deprecated. However, the next release will be a major version bump that will require people to make some changes in their app.